background/services/signing/index.ts

```diff
@@ -37,19 +54,64 @@ export default class SigningService extends BaseService<Events> {
     await super.internalStartService() // Not needed, but better to stick to the patterns
   }
 
-  async deriveAddress(accountID: string): Promise<HexString> {
-    this.deriveAddress = this.deriveAddress.bind(this)
+  async deriveAddress(signerID: string): Promise<HexString> {
+    if (signerID.startsWith("ledger-")) {
+      return this.ledgerService.deriveAddress(
+        signerID.substring("ledger-".length)
+      )
+    }
 
-    throw new Error("Unimplemented")
+    if (signerID.startsWith("keyring-")) {
+      return this.keyringService.deriveAddress(
+        signerID.substring("keyring-".length)
+      )
+    }
+
+    throw new Error(`Unknown signerID: ${signerID}`)
   }
 
   async signTransaction(
     address: HexString,
     transactionRequest: EIP1559TransactionRequest
   ): Promise<SignedEVMTransaction> {
-    this.signTransaction = this.signTransaction.bind(this)
+    const transactionWithNonce =
+      await this.chainService.populateEVMTransactionNonce(transactionRequest)
 
-    throw new Error("Unimplemented")
+    try {
+      const actualHandler = this.addressHandlers.find(
+        (handlers) => handlers.address === address
+      )
+
+      if (!actualHandler) {
+        throw new Error(`Unregistered address (${address}) found!`)
+      }
+
+      switch (actualHandler.handler) {
+        case "ledger":
+          return await this.ledgerService.signTransaction(
+            transactionRequest.from,
+            transactionWithNonce
+          )
+        case "keyring":
+          return await this.keyringService.signTransaction(
+            transactionRequest.from,
+            transactionWithNonce
+          )
+        default:
+          throw new Error(
+            `Unknown address (${address}) or handler (${actualHandler}) provided!`
+          )
+      }
+    } catch (exception) {
+      logger.error("Error signing transaction; releasing nonce", exception)
+      this.chainService.releaseEVMTransactionNonce(transactionWithNonce)
+    }
+
+    throw new Error("Impossible to reach")
```

Comment:
```suggestion
    } finally {
      this.chainService.releaseEVMTransactionNonce(transactionWithNonce)
    }
```

No point in catching the original exception and re-throw our own.
I assume the logging is done elsewhere anyway. If not, then it should be added higher up in the call hirearchy.